internal/store/parent_subnet.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+package store
+
+import (
+	"database/sql"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/mattermost/genesis/model"
+	"github.com/pkg/errors"
+)
+
+var parentSubnetSelect sq.SelectBuilder
+
+func init() {
+	parentSubnetSelect = sq.
+		Select("ParentSubnet.ID", "CIDR", "SplitRange", "CreateAt",
+			"LockAcquiredBy", "LockAcquiredAt").
+		From("ParentSubnet")
+}
+
+type rawParentSubnet struct {
+	*model.ParentSubnet
+}
+type rawParentSubnets []*rawParentSubnet
+
+func (r *rawParentSubnet) toParentSubnet() (*model.ParentSubnet, error) {
+	return r.ParentSubnet, nil
+}
+
+func (rc *rawParentSubnets) toParentSubnets() ([]*model.ParentSubnet, error) {
+	var parentSubnets []*model.ParentSubnet
+	for _, rawParentSubnet := range *rc {
+		parentSubnet, err := rawParentSubnet.toParentSubnet()
+		if err != nil {
+			return nil, err
+		}
+		parentSubnets = append(parentSubnets, parentSubnet)
+	}
+
+	return parentSubnets, nil
+}
+
+// GetParentSubnet fetches the given parent subnet by subnet range.
+func (sqlStore *SQLStore) GetParentSubnet(id string) (*model.ParentSubnet, error) {
+	var rawParentSubnet rawParentSubnet
+	err := sqlStore.getBuilder(sqlStore.db, &rawParentSubnet, parentSubnetSelect.Where("ID = ?", id))
+	if err == sql.ErrNoRows {
+		return nil, nil
```

Comment:
I see we rely a lot in nullability which can make the app to crash with panic and adds a bit more extra ifs. I have seen a few errors in our logs too are `panic: runtime error: invalid memory address`. 

Not sure what we follow in mattermost for golang but it's a nice approach to return or a custom error eg. `NotFound` or even the same error with an empty struct `model.ParentSubnet{}` which is also a good approach to avoid nils and you rely in the idiomatic go check the error first etc.